runtime/js/mlBytes.js

```diff
@@ -78,7 +78,7 @@ function caml_subarray_to_jsbytes(a, i, len) {
   if (i === 0 && len <= 4096 && len === a.length) return f.apply(null, a);
   var s = "";
   for (; 0 < len; i += 1024, len -= 1024)
-    s += f.apply(null, a.slice(i, i + Math.min(len, 1024)));
+    s += f.apply(null, a.subarray(i, i + Math.min(len, 1024)));
```

Comment:
It is not clear that `caml_subarray_to_jsbytes`  is not used on `Array`.  How do you feel about introducing a new function and updating the callers ? `caml_sub_int8_array_to_jsbytes` ? We would need another function for the `caml_string_of_array` counterpart. 

Edit: It seems that BigStringReader.readstr is indirectly calling caml_subarray_to_jsbytes with an `Array`, so is `caml_ml_output_int`